sched/pthread/pthread_create.c

```diff
@@ -442,19 +442,19 @@ int nx_pthread_create(pthread_trampoline_t trampoline, FAR pthread_t *thread,
 #endif
     }
 
-  /* Then activate the task */
-
   sched_lock();
 
-  nxtask_activate((FAR struct tcb_s *)ptcb);
-
   /* Return the thread information to the caller */
 
   if (thread != NULL)
     {
       *thread = (pthread_t)ptcb->cmn.pid;
     }
 
+  /* Then activate the task */
+
+  nxtask_activate((FAR struct tcb_s *)ptcb);
+
   sched_unlock();
```

Comment:
Isn't here protected by sched_lock?

Reply:
already in sched_lock

Reply:
so why does the context switch happen? Won't it be delayed until sched_unlock()?

Reply:
Emm, sorry, I have misunderstood the intention of this PR